### PR TITLE
[RUM-5590] Add telemetry for INP null target

### PR DESCRIPTION
**Pull request by cy-moi**
## Motivation
Add telemetry for INP null target to get the impact.

## Changes
Add telemetry debug 

## Testing
Screenshot 2024-07-26 at 09 20 45

- [x] Local
- [ ] Staging
- [ ] Unit
- [ ] End to end

---

I have gone over the [contributing](https://github.com/DataDog/browser-sdk/blob/main/CONTRIBUTING.md) documentation.

### CODE REVIEW

File: packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts [hunk 1]
```diff
@@ -74,6 +73,10 @@ export function trackInteractionToNextPaint(
           configuration.actionNameAttribute
         )
       } else {
+        addTelemetryDebug('INP target is null or not an element node', {
+          target: JSON.stringify(newInteraction.target), // if target is not an element node, log it to help debugging
```
Comment (amortemousque) [inline packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts:77]: **🔨 warning:** JSON.stringify(newInteraction.target) will always return '{}'. Maybe you could do it like so:
```ts
{
hasTarget: !!newInteraction.target
targetIsConnected: newInteraction.target?.isConnected,
...
}
```
‏

**Review by N-Boutaib — APPROVED** [review]

File: packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts [hunk 2]
```diff
@@ -65,14 +65,18 @@ export function trackInteractionToNextPaint(
...
+      const inpTarget = newInteraction.target
       interactionToNextPaint = newInteraction.duration
       interactionToNextPaintStartTime = elapsed(viewStart, newInteraction.startTime)
 
-      if (newInteraction.target && isElementNode(newInteraction.target)) {
-        interactionToNextPaintTargetSelector = getSelectorFromElement(
-          newInteraction.target,
-          configuration.actionNameAttribute
-        )
+      addTelemetryDebug('INP target is null or not an element node', {
```
Comment (amortemousque) [inline packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts:72]: **💬 suggestion:** ‏To reduce the number of telemetry, you could collect it only when there is no `interactionToNextPaintTargetSelector`.
Also collecting the telemetry on org2 could be enough. So you could use a feature flag.

**Review by N-Boutaib — APPROVED** [review]
LGTM

File: packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts [hunk 3]
```diff
@@ -65,15 +72,28 @@ export function trackInteractionToNextPaint(
...
+      const inpTarget = newInteraction.target
       interactionToNextPaint = newInteraction.duration
       interactionToNextPaintStartTime = elapsed(viewStart, newInteraction.startTime)
-
-      if (newInteraction.target && isElementNode(newInteraction.target)) {
-        interactionToNextPaintTargetSelector = getSelectorFromElement(
-          newInteraction.target,
-          configuration.actionNameAttribute
-        )
+      const isInpEnabled = isExperimentalFeatureEnabled(ExperimentalFeature.NULL_INP_TELEMETRY)
```
Comment (thomas-lebeau) [inline packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts:78]: **🥜 nitpick:** ‏ shouldn't it be `isInpTelemetryEnabled`?

**Review by thomas-lebeau — APPROVED** [review]

File: packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts [hunk 4]
```diff
@@ -65,15 +72,28 @@ export function trackInteractionToNextPaint(
...
       } else {
+        if (isInpEnabled) {
+          addTelemetryDebug('INP target is null or not an element node', {
+            hasTarget: !!inpTarget,
+            targetIsConnected: inpTarget ? inpTarget.isConnected : null,
+            inp: newInteraction.duration,
+          })
+        }
+
         interactionToNextPaintTargetSelector = undefined
```
Comment (BenoitZugmeyer) [inline packages/rum-core/src/domain/view/viewMetrics/trackInteractionToNextPaint.ts]: **💬 suggestion:** ‏to simplify a little bit:

```ts
      if (inpTarget && isElementNode(inpTarget)) {
        interactionToNextPaintTargetSelector = getSelectorFromElement(inpTarget, configuration.actionNameAttribute)
      } else {
        interactionToNextPaintTargetSelector = undefined
      }
      if (!interactionToNextPaintTargetSelector && isExperimentalFeatureEnabled(ExperimentalFeature.NULL_INP_TELEMETRY)) {
         addTelemetryDebug("Fail to get INP target selector", {
           hasTarget: !!inpTarget,
           targetIsConnected: inpTarget ? inpTarget.isConnected : undefined,
           targetIsElementNode: inpTarget ? isElementNode(inpTarget) : undefined,
           inp: newInteraction.duration
         })
      }
```

Also collect `isElementNode` so we cover the whole condition (`inpTarget && isElementNode(inpTarget)`)

**Pull request merged**